Replace user with security.principal in hdfs repo (#7876)
crate/crate: blackbox/docs/appendices/release-notes/unreleased.rst

- Added support of using units inside byte-size or time bases statement
  parameters values. E.g. '1mb' for 1 MegaByte or '1s' for 1 Second.

> - Changed hdfs repository parameter, ``user``, to ``security.principal`` so

Reviewer: ```suggestion
- Changed the ``user`` hdfs repository parameter to ``security.principal`` as that is
the parameter used by the underlying repository-hdfs plugin.
```

> - Changed hdfs repository parameter, ``user``, to ``security.principal`` so

Reviewer: also, super minor, but we generally add the newer changes at the top. can you please move them just above "Added support for CrateDB license management ..."

  as to come in line with the underlying repository-hdfs plugin.

> - Removed hdfs repository parameter, ``conf_location``, as it appears to no

Reviewer: ```suggestion
- Removed the ``conf_location`` hdfs repository parameter as it is not used anymore.
```
